atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -351,6 +359,16 @@ private void addSnapshotListener() {
                 .build()));
   }
 
+  private void onSnapshotReceiveCompletedOrAborted() {
+    if (pendingSnapshot != null) {
+      pendingSnapshot.abort();
```

Comment:
:question: Why are we aborting the snapshot here? From the name I would expect this to already have been done, i.e. either persisted or aborted prior to this call. From reading the code, it seems like that's because we call this if we receive an append request while replicating a snapshot. Why not call `abortPendingSnapshots`?

Reply:
We can call `abortPendingSnapshots`, followed by `onSnapshotReceiveCompletedOrAborted()` when receiving append request.